packages/@uppy/dashboard/src/components/FileItem/index.js

```diff
@@ -1,76 +1,89 @@
-const { h } = require('preact')
+const { h, Component } = require('preact')
 const classNames = require('classnames')
-const pure = require('../../utils/pure')
+const shallowEqual = require('is-shallow-equal')
 const FilePreviewAndLink = require('./FilePreviewAndLink')
 const FileProgress = require('./FileProgress')
 const FileInfo = require('./FileInfo')
 const Buttons = require('./Buttons')
 
-module.exports = pure(function FileItem (props) {
-  const file = props.file
+module.exports = class FileItem extends Component {
+  shouldComponentUpdate (nextProps) {
+    return !shallowEqual(this.props, nextProps)
```

Comment:
Same than `pure`, but a 3-rd party module?

Reply:
this basically inlines the implementation of `pure()` here. previously `FileItem` was a function component, so to avoid turning it into a class component and bloating the diff, I put this shallowEqual check in a higher order component `pure()`. but now it's already a class component because we need the `componentDidMount()` hook, and i think it's a bit overkill to have a HOC that's only used in one place